### ISSUE TITLE
Conda dependency baseos for ctsm-fates in NorESM Galaxy project

### ISSUE THREAD

**Issue by glemieux** [NGEET/docker-fates-tutorial#30]
The [Nordic Earth System Model (NorESM)](https://github.com/NordicESMhub) is integrating FATES and working on utilizing its model within the [Galaxy project](https://github.com/galaxyproject): https://github.com/NordicESMhub/galaxy-tools/issues/39.  Since the Galaxy project makes use of containers to promote reproducible science and cloud computing, we are collaborating with NorESM to bring containerized hlm-fates to Galaxy.  The hope is that this will help further experience with and adoption of the FATES model.  

Galaxy utilizes conda's package management infrastructure to distribute and maintain containers, specifically [bioconda containers](https://github.com/BioContainers/containers).  As such, it is ideal that the dockerfile recipes utilize conda-based libraries for the containerized application dependencies.

**Comment by glemieux** [comment]
Attempted to adopt the galaxy workflow of automating the container build.  Currently unable to get the local tests to run.  Following this documentation: https://docs.galaxyproject.org/en/latest/admin/special_topics/mulled_containers.html#building-docker-containers-for-local-conda-packages

My [test PR](https://github.com/conda-forge/staged-recipes/pull/12972) submitted to conda-forge failed a check.  PR is based off of https://github.com/annefou/staged-recipes/tree/fates/recipes/fates.  Need to review the failure and figure out next steps.  This is based off of the

**Comment by glemieux** [comment]
Looking at alternative method of providing existing dockerfile recipe to fit within the Galaxy workflow if possible.

**Comment by annefou** [comment]
Yes I have seen this kind of error; right now I can't remember exactly why and when but i will look at it.

**Comment by bgruening** [comment]
@glemieux your conda package failed because it does not work under python3 can this be?

**Comment by annefou** [comment]
The python errors in statistical_ensemble_test should not be a problem (I mean not for the conda build). The license is missing:
` File "/opt/conda/lib/python3.8/site-packages/conda_build/build.py", line 783, in copy_license
    raise ValueError("License file given in about/license_file ({}) does not exist in "
ValueError: License file given in about/license_file (/home/conda/staged-recipes-copy/recipes/ctsm-fates/LICENSE.txt) does not exist in source root dir or in recipe root dir (with meta.yaml)

`

**Comment by annefou** [comment]
in meta.yaml:

`license_file: LICENSE
`

While now you have

`license_file: LICENSE.txt`

And this is my fault because I made this error and corrected it later for fates-emerald.  The conda-forge recipe for fates-emerald is at https://github.com/conda-forge/fates-emerald-feedstock

**Comment by bgruening** [comment]
But this is also hinting at a python2 problem: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=230158&view=logs&j=10bda42a-d420-5c28-a43f-bc4e66405e5c&t=bef775d0-a1a2-55a3-9721-17147029baf0&l=885

**Comment by annefou** [comment]
Yes, I also have it in my conda recipe... It passes even with these errors (this part is not used; additional tools still in python 2; I mean for print statements... Could easily be migrated to python 3; i am not sure who would be using these).

**Comment by glemieux** [comment]
Sorry for the late replies.  Thanks for all the feedback.  I'm guessing that this is low-hanging fruit that the CIME team just hasn't prioritized due to more pressing efforts.  My guess is that they have to support python 2 by default on their supported machines anyway so their isn't a huge impetus to fix this.  I'll double check with them though.

**Comment by glemieux** [comment]
Update: https://github.com/ESMCI/cime/issues/3776
